osmnx/utils_graph.py

```diff
@@ -82,9 +81,7 @@ def make_geom(u, v, data, x=x_lookup, y=y_lookup):
             gdf_edges = gpd.GeoDataFrame(data)
             if "geometry" not in gdf_edges.columns:
                 # if no edges have a geometry attribute, create null column
-                gdf_edges["geometry"] = np.nan
-            gdf_edges.set_geometry("geometry")
-            gdf_edges.crs = crs
+                gdf_edges = gdf_edges.set_geometry([None] * len(gdf_edges), crs=crs)
```

Comment:
The CI tests are failing because gdf_edges only gets a `crs` attribute if this if statement is True. And we can't pass crs into the constructor on line 81 because it fails when there is no geometry in `data`. Does a change like this seem appropriate?
```python
gdf_edges = gpd.GeoDataFrame(data)
if "geometry" not in gdf_edges.columns:
    # if no edges have a geometry attribute, create null column
    gdf_edges = gdf_edges.set_geometry([None] * len(gdf_edges))
gdf_edges.crs = crs
```
I suppose my broader question is: what's the appropriate GeoPandas technique nowadays for creating a GeoDataFrame that *may* or *may not* have geometry in the data we pass to the constructor (such as in this case)? In this proposed change we 1) create the GeoDataFrame, 2) check if a named "geometry" column exists, 3) if not, then create a null "geometry" column, then 4) set the crs. Is this right?

Reply:
This is a bit of a corner case but I suppose this would be a correct technique. With a minor suggestion to use `set_crs` instead of assigning crs to the attribute as there are some additional checks in `set_crs`.

For this use cases, it may be useful to have an attribute on a GeoDataFrame that gives you an information of an active geometry column is set or not, something like `GeoDataFrame.has_geometry` but we don't have that now (will open an issue).

Reply:
Great, thank you.